From: Andrej Shadura <andrew.shadura@collabora.co.uk>
Date: Tue, 29 Jan 2019 10:29:21 +0100
Subject: [PATCH] debian/rules: redact the build directory out of the tarball paths

When the tarball for binutils-source package is being created, the
directory name where the package is unpacked may be captured by tar
and embedded into the tarball paths. This may cause further inconvenience
when unpacking the tarball. One of the Debian packages in affected by
this: binutils-mingw64 fails to unpack binutils tarball.

To mitigate the issue, replace the binutils-source tarball path prefix with
binutils-$(VERSION).

Signed-off-by: Andrej Shadura <andrew.shadura@collabora.co.uk>
---

diff --git a/debian/rules b/debian/rules
--- a/debian/rules
+++ b/debian/rules
@@ -1336,6 +1336,7 @@
 		xargs -0r touch --no-dereference --date='$(BUILD_DATE)' && \
 		find $(source_files) -type f -print0 | LC_ALL=C sort -z | \
 		tar --null -T - -c --xz --exclude=CVS --mode=go=rX,u+rw,a-s \
+		--xform='s=^[^/]*\/=binutils-$(VERSION)/=' \
 		-f $(pwd)/$(d_src)/$(PF)/src/binutils/binutils-$(VERSION).tar.xz \
 		$(source_files)
 
